lib/alex_koin/commands/destroy_koin.ex

```diff
@@ -0,0 +1,45 @@
+defmodule AlexKoin.Commands.DestroyKoin do
+  require Logger
+  alias AlexKoin.Repo
+  alias AlexKoin.SlackCommands
+  alias AlexKoin.SlackDataHelpers
+  alias AlexKoin.Account.User
+  alias AlexKoin.Coins.Coin
+
+  def execute(user, _message, text, slack, amount \\ 1) do
+    cond do
+      User.admin?(user) ->
+        do_destroy_koin(user, amount, text, slack)
+
+      User.koin_lord?(user) ->
+        do_destroy_koin(user, amount, text, slack)
+
+      true ->
+        peasant_message = "You pathetic human you don't have any family any friends or any land"
+        SlackDataHelpers.message_ts(peasant_message)
+    end
+  end
+
+  defp do_destroy_koin(user, amount, text, slack) do
+    regex = ~r/<@(?<to_slack_id>[A-Z0-9]+)> for (?<reason>.*)/
+    if Regex.match?(regex, text) do
```

Comment:
So this is a good time to learn about pattern matching in Elixir! It's typically a code smell to ever use an if statement in Elixir because of pattern matching.

This bit where it does the regex match, should actually be a function that uses a Guard Clause: https://hexdocs.pm/elixir/guards.html

Reply:
This is cool, but it looks like Regex is not allowed as a method to be used for guards: 
```
cannot invoke remote function Regex.match?/2 inside guard
```
https://stackoverflow.com/questions/34850121/why-forbidden-to-use-a-remote-function-inside-a-guard